python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -255,6 +267,30 @@ def canFetchMore(self, index):
     ############################################################################
     # protected methods
 
+    def _check_hierarchy_is_supported(self):
+        """
+        Checks the current Shotgun connection to make sure it supports
+        hierarchy queries.
+        """
+
+        # get the current shotgun connection
+        current_engine = sgtk.platform.current_engine()
+        sg_connection = current_engine.shotgun
+        server_caps = sg_connection.server_caps
+
+        if not hierarchy_is_supported(sg_connection):
+            # oops, SG version is not compatible
+            cur_sg_version = "v%s" % (
+                ".".join(map(str,server_caps.version)))
+            logger.warning(
```

Comment:
this warning you are composing here could be the exception message raised from the SG API...